spotbugsTestCases/src/java/ghIssues/Issue705.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Contributions to SpotBugs
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+package ghIssues;
+
+public class Issue705 {
+    
+    public static String bug() {
+        String replaceable = "replaceable";
+        String replacement = "replacement";
+        String target = "target";
+        return target.replaceAll(replaceable, replacement);
+    }
+
+    public static String nonBug() {
+        String replaceable = "(replaceable)";
+        String replacement = "replacement";
+        String target = "target";
+        return target.replaceAll(replaceable, replacement);
+    }
+    
+    public static String nonBug2() {
```

Comment:
Nice coverage :+1: 

Might be easier to make an array of the metacharacters and loop through that.

Reply:
The problem with that might be that the arguments are no longer recognized as constants. I don't know what the full capabilities of Spotbugs in those situations are.

Reply:
Oh, of course. You're right, it would probably fail to trip the detector for the wrong reasons.

I shouldn't try to make bug test cases elegant :laughing: